### DISCUSSION TITLE
Time of use select options

### DISCUSSION THREAD

**Discussion by wepster** [davidrapan/ha-solarman#333] [General]
For "time of use" there are many options: disabled, enabled, weekdays, weekend, week and every day of week.
What all of them mean? I don't see analog options when usuing inverter's screen to configure. There "time of use" is a checkbox.
I would like to set different times of use for weekdays and weekends, is this the way to do this? I am a bit affraid to experiment with this.

**Comment by sofkaski** [comment]
I have understood that in Deye you can choose whether use TOU at all or then a defined same time program on selected days.
 
I have ended up using this so that I use TOU on all 7 days and modify the time program with automation on the need basis. Currently my automation selects an optimal time to charge the battery from the grid based on hourly spot prices. For consumption part automation tries to use battery when the prices are highest. There I use the battery SOC % to control battery discharge. You can use also allowed power as the controlling mechanism, but I have found it easier to use SOC %. In my setup the inverter gets the SOC % from the battery management system. (The inverter and the battery pack are connected with CAN bus).

**Reply by sofkaski** [reply]
I have this kind of overview of programs (markdown card):

![image](https://github.com/user-attachments/assets/fb736b07-3fbf-4ad1-b49c-9874cd382e96)

So here battery was charged starting at 4:00 to full capacity, but battery is not used today at all (cheap electricity price here due to windy weather).

**Reply by wepster** [reply]
> I have understood that in Deye you can choose whether use TOU at all or then a defined same time program on selected days.

That is reasonable, i will test that when i get the chance, problably next year though
